app/views/helpers/FullResultsPageHelper.scala

```diff
@@ -366,4 +367,89 @@ object FullResultsPageHelper extends ViewHelper {
+              TableRow(content = HtmlContent(p(messages("fullResultsPage.taxableProfit")))),
+              TableRow(
+                content = HtmlContent(CurrencyUtils.decimalFormat(d1.adjustedProfit)),
+                classes = "govuk-table__cell--numeric"
+              ),
+              TableRow(
+                content = HtmlContent(CurrencyUtils.decimalFormat(BigDecimal(d2.adjustedProfit))),
+                classes = "govuk-table__cell--numeric"
+              ),
+              TableRow(
+                content = HtmlContent(CurrencyUtils.decimalFormat(taxableProfit)),
+                classes = "govuk-table__cell--numeric"
+              )
+            ),
+            Seq(
+              TableRow(content = HtmlContent(p(messages("fullResultsPage.taxableProfit.distributions")))),
+              TableRow(
+                content = HtmlContent(CurrencyUtils.decimalFormat(d1.adjustedDistributions)),
+                classes = "govuk-table__cell--numeric"
+              ),
+              TableRow(
+                content = HtmlContent(CurrencyUtils.decimalFormat(d2.adjustedDistributions)),
+                classes = "govuk-table__cell--numeric"
+              ),
+              TableRow(
+                content = HtmlContent(CurrencyUtils.decimalFormat(distributions)),
+                classes = "govuk-table__cell--numeric"
+              )
+            ),
+            Seq(
+              TableRow(content = HtmlContent(p(messages("fullResultsPage.taxableProfit.profitAndDistributions")))),
+              TableRow(
+                content = HtmlContent(CurrencyUtils.decimalFormat(d1.adjustedAugmentedProfit)),
+                classes = "govuk-table__cell--numeric"
+              ),
+              TableRow(
+                content = HtmlContent(CurrencyUtils.decimalFormat(d2.adjustedAugmentedProfit)),
+                classes = "govuk-table__cell--numeric"
+              ),
+              TableRow(
+                content = HtmlContent(CurrencyUtils.decimalFormat(taxProfitDistributions)),
+                classes = "govuk-table__cell--numeric"
+              )
+            )
+          ),
+          head = Some(
+            Seq(
+              HeadCell(),
+              HeadCell(content = Text(messages("site.from.to", d1.year.toString, (d1.year + 1).toString))),
+              HeadCell(content = Text(messages("site.from.to", d2.year.toString, (d2.year + 1).toString))),
+              HeadCell(content = Text(messages("fullResultsPage.total")))
+            )
+          ),
+          caption = Some(messages("fullResultsPage.taxableProfit")),
+          captionClasses = "govuk-table__caption--m"
+        )
+      )
+    }
+
+    calculatorResult.fold(single => HtmlFormat.empty)(dual =>
+      dual.year1 -> dual.year2 match {
+        case (d1: FlatRate, d2: MarginalRate) => table(d1, d2)
+        case (d1: MarginalRate, d2: FlatRate) => table(d1, d2)
+        case _                                => HtmlFormat.empty
+      }
```

Comment:
Isnt there is taxable profit table for Single(marginalrate) and Dual(marginalrate, marginalrate) scenario?

Reply:
No, I think the logic here is right. This table is only displayed in a very specific scenario.

Reply:
Cool, looks good!